content/doc/pipeline/index.adoc

```diff
@@ -1,981 +1,437 @@
 
-The CloudBees Jenkins Platform has a robust template feature. CloudBees Jenkins Platform users can create template build steps, jobs, folders, and publishers. Since Pipelines are a new job type, authors can create Pipeline templates so that similar pipelines can simply leverage the same Pipeline job template. More information on Templates is available on the CloudBees' website:
+ node {
+   git url: 'https://github.com/joe_user/simple-maven-project-with-tests.git'
+   def mvnHome = tool 'M3'
+   sh "${mvnHome}/bin/mvn -B -Dmaven.test.failure.ignore verify"
+   step([$class: 'ArtifactArchiver', artifacts: '**/target/*.jar', fingerprint: true])
+   step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+ }
 
-https://www.cloudbees.com/products/cloudbees-jenkins-platform/enterprise-edition/features/templates-plugin
+* If tests fail, the Pipeline is marked unstable (as denoted by a yellow ball in the Jenkins UI), and you can browse the Test Result Trend to see the involved history.
+* You should see Last Successful Artifacts on the Pipeline index page.
 
-# Tying it Together: Example Pipeline
 
-The following pipeline is an example tying together several of the Pipeline features we learned earlier. While not exhaustive, it provides a basic but complete pipeline that will help jump-start your pipeline development:
+== Loading Pipeline Scripts from SCM
+Complex pipelines would be cumbersome to write and maintain if you could only do that in the text area provided by the Jenkins job configuration page. 
 
-```groovy
-stage 'build'
-node {
-   git 'https://github.com/cloudbees/todo-api.git'
-   withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
-      sh "mvn -B –Dmaven.test.failure.ignore=true clean package"
-   }
-   stash excludes: 'target/', includes: '**', name: 'source'
-}
-stage 'test'
-parallel 'integration': {
-   node {
-      unstash 'source'
-      withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
-         sh "mvn clean verify"
-      }
-   }
-}, 'quality': {
-   node {
-      unstash 'source'
-      withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
-         sh "mvn sonar:sonar"
-      }
-   }
-}
-stage 'approve'
-timeout(time: 7, unit: 'DAYS') {
-   input message: 'Do you want to deploy?', submitter: 'ops'
-}
-stage name:'deploy', concurrency: 1
-node {
-   unstash 'source'
-   withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
-     sh "mvn cargo:deploy"
-   }
-}
-```
+Accordingly, you also have the option of writing pipeline scripts in in your IDE (integrated development environment) or SCM system, and then loading those scripts into Jenkins using the *Pipeline Script from SCM* option enabled by the workflow-scm-step plugin, which is one of the plugins that the Pipeline plugin depends on and automatically installs.
```

Comment:
"writing pipeline scripts in in"

one too many *in* words here

----

Also I don't write pipeline scripts in my "SCM system", but I think I follow what is trying to be communicated here. I write my Jenkisnfiles in my editor and check them into my SCM system.

---

> option enabled by the workflow-scm-step plugin

This is confusing and unnecessary detail in this kind of documentation. Since the "Pipeline Script from SCM" is installed when you install the "Pipeline" plugin, I think this statement can be removed.